ui/components/src/model/legend.ts

```diff
@@ -28,7 +31,7 @@ export interface LegendItem {
   onClick?: MouseEventHandler<HTMLElement>;
 }
 
-export type LegendPositionConfig = {
+export type LegendSingleSelectConfig = {
```

Comment:
Renaming this to be relatively generic, so it can also be used with the `mode`.

Reply:
What will the proposed JSON spec / CUE change be in the next PR, sounds like a new legend.mode or possibly `legend.kind`to switch between List and Table?

Reply:
I was going with `legend.mode`. You can see a work-in-progress here: https://github.com/perses/perses/pull/1197/files#diff-b9ac1aa40a10aa0045374ee90f440e262d2c8f11c2a0292bccdb2d78e420d355